cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java

```diff
@@ -589,6 +600,20 @@ private Future<Void> podSet(int replicas) {
         }
     }
 
+    /**
+     * Prepares annotations for ZooKeeper pods within a StrimziPodSet.
+     *
+     * @param podNum Number of the ZooKeeper pod, the annotations of which are being prepared.
+     * @return Map with Pod annotations
+     */
+    public Map<String, String> zkPodSetPodAnnotations(int podNum) {
+        Map<String, String> podAnnotations = new LinkedHashMap<>(2);
```

Comment:
To avoid inevitable reallocation I guess you need to change the initial capacity:

```suggestion
        Map<String, String> podAnnotations = new LinkedHashMap<>(1 + (int)(3/0.75));
```